webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/logger/DebugComplianceLogger.scala

```diff
@@ -86,6 +87,48 @@ object ComplianceDebugLogger extends Logger {
 
   def node(id: NodeId): Logger = nodeCache.get(id.value)
 
+  implicit class RunAnalysisInfoToLog(val c: RunAnalysis) extends AnyVal {
+    import com.normation.rudder.domain.reports.RunAnalysisKind.*
+
+    def logId:  String = {
+      s"${c.expectedConfigId.fold("unknown id")(_.value)}/[${c.expectedConfigStart.fold("unknown start time")(
+          _.toIsoStringNoMillis
+        )}-${c.expirationDateTime.fold("unknown expiration date")(_.toIsoStringNoMillis)}]"
+    }
+    def logRun: String = {
+      s"${c.lastRunConfigId.fold("none or too old")(i => s"configId: ${i.value}")} received at: ${c.lastRunDateTime.fold("unknown")(
+          _.toIsoStringNoMillis
+        )}${c.lastRunExpiration.fold("")(r => s" | expired at ${r.toIsoStringNoMillis}")}"
+    }
+
+    def logDetails: String = c.kind match {
+      case NoRunNoExpectedReport =>
+        "expected NodeConfigId: not found | last run: not found"
+
+      case NoExpectedReport =>
+        s"expected NodeConfigId: not found |" +
+        s" last run: nodeConfigId: ${c.lastRunConfigId.fold("none")(_.value)} received at ${c.lastRunDateTime.fold("undef")(_.toIsoStringNoMillis)}"
+
+      case NoReportInInterval =>
+        s"expected NodeConfigId: ${c.logId}|" +
+        s" last run: none available (or too old)"
+
+      case KeepLastCompliance =>
+        s"expected NodeConfigId: ${c.logId}|" +
+        s" last: ${c.logRun} ; expired but will be kept until ${c.expirationDateTime.fold("unknown")(_.toIsoStringNoMillis)}"
+
+      case ReportsDisabledInInterval =>
+        s"expected NodeConfigId: ${c.logId}|" +
+        s" last run: none available (compliance mode is reports-disabled)]"
+
+      case _ =>
+        s"expected NodeConfigId: ${c.logId} |" +
+        s" last run: ${c.logRun}"
```

Comment:
The last one has a space before `|`, all the other ones don't, which make regex harder than they already are... :innocent: